pylint/checkers/base.py

```diff
@@ -386,36 +387,49 @@ def _has_abstract_methods(node):
     return len(utils.unimplemented_abstract_methods(node)) > 0
 
 
-def report_by_type_stats(sect, stats, old_stats):
+def report_by_type_stats(
+    sect,
+    stats: CheckerStatistics,
+    old_stats: CheckerStatistics,
+):
     """make a report of
 
     * percentage of different types documented
     * percentage of different types with a bad name
     """
     # percentage of different types documented and/or with a bad name
-    nice_stats = {}
+    nice_stats: Dict[str, Dict[str, str]] = {}
     for node_type in ("module", "class", "method", "function"):
         try:
-            total = stats[node_type]
+            total: int = stats[node_type]  # type: ignore
         except KeyError as e:
             raise exceptions.EmptyReportError() from e
         nice_stats[node_type] = {}
         if total != 0:
             try:
-                documented = total - stats["undocumented_" + node_type]
+                undocumented_node: int = stats["undocumented_" + node_type]  # type: ignore
+                documented = total - undocumented_node
                 percent = (documented * 100.0) / total
                 nice_stats[node_type]["percent_documented"] = f"{percent:.2f}"
             except KeyError:
                 nice_stats[node_type]["percent_documented"] = "NC"
             try:
-                percent = (stats["badname_" + node_type] * 100.0) / total
+                badname_node: int = stats["badname_" + node_type]  # type: ignore
+                percent = (badname_node * 100.0) / total
                 nice_stats[node_type]["percent_badname"] = f"{percent:.2f}"
             except KeyError:
                 nice_stats[node_type]["percent_badname"] = "NC"
-    lines = ("type", "number", "old number", "difference", "%documented", "%badname")
+    lines: Tuple[str, ...] = (
```

Comment:
This should be typed implicitly.

Reply:
The problem is on line 437. Because the length of the tuple changes there. `mypy` doesn't like that and complains about a difference in typing. I could add an ignore on that line (and on the other cases you mentioned), but `Tuple[str, ...]` might be better in this case.

Reply:
If additional items are added later, wouldn't it be better if lines where a `List[str]` instead?

Reply:
Yes! I can make all occurrences of `lines` a list and type them as `List[str]`. Want me to do that?
Originally I didn't want to change any code if not necessary and only add typing.

Reply:
I think that would make sense. Could you do that in a separate PR? Just the change to `list[str]`. That should be easy to review and fast to merge.

Reply:
Yes, give me a couple of minutes!

Reply:
Tag me, and I'll take a look.

Reply:
See https://github.com/PyCQA/pylint/pull/5009